networkx/algorithms/summarization.py

```diff
@@ -0,0 +1,187 @@
+"""
+Algorithms for reducing the number of nodes/edges in a graph
+
+*dedensification* reduces the number of edges in unweighted graphs by adding
```

Comment:
Could you say something about what properties this graph summary preserves.  It obviously changes the number of nodes and edges.  It changes the node degrees.  It also changes things like the number of distinct edges paths.  What does it preserve?  What types of algorithms can you still run on it (and get an answer that tells you something about the original graph)?

Reply:
Yes, I will add this.

Reply:
I have given this a first attempt.  I was not sure how detailed to get with it, so let me know what needs to be changed

Reply:
@jarrodmillman when you get a chance, could you review the module documentation?